core/controller/src/main/scala/whisk/core/controller/Triggers.scala

```diff
@@ -305,59 +292,56 @@ trait WhiskTriggersApi extends WhiskCollectionAPI {
   private def activateRules(user: Identity,
                             args: JsObject,
                             rulesToActivate: Map[FullyQualifiedEntityName, ReducedRule])(
-    implicit transid: TransactionId): Iterable[Future[JsObject]] = {
-    rulesToActivate.map {
+    implicit transid: TransactionId): Future[Iterable[RuleActivationResult]] = {
+    val ruleResults = rulesToActivate.map {
       case (ruleName, rule) =>
         // Invoke the action. Retain action results for inclusion in the trigger activation record
-        val actionActivationResult: Future[JsObject] = postActivation(user, rule, args)
+        postActivation(user, rule, args)
           .flatMap { response =>
             response.status match {
               case OK | Accepted =>
                 Unmarshal(response.entity).to[JsObject].map { activationResponse =>
-                  val activationId: JsValue = activationResponse.fields("activationId")
+                  val activationId = activationResponse.fields("activationId").convertTo[ActivationId]
                   logging.debug(this, s"trigger-fired action '${rule.action}' invoked with activation $activationId")
-                  ruleResult(ActivationResponse.Success, ruleName, rule.action, Some(activationId))
+                  RuleActivationResult(ActivationResponse.Success, ruleName, rule.action, Right(activationId))
                 }
 
-              // all proper controller responses are JSON objects that deserialize to an ErrorResponse instance
-              case code if (response.entity.contentType == ContentTypes.`application/json`) =>
-                Unmarshal(response.entity).to[ErrorResponse].map { e =>
-                  val statusCode =
-                    if (code != InternalServerError) {
-                      logging
-                        .debug(
-                          this,
-                          s"trigger-fired action '${rule.action}' failed to invoke with ${e.error}, ${e.code}")
-                      ActivationResponse.ApplicationError
-                    } else {
+              case code =>
+                Unmarshal(response.entity).to[String].map { error =>
+                  val failureType = code match {
+                    case _: ServerError => ActivationResponse.WhiskError // all 500s are to be considered whisk errors
+                    case _              => ActivationResponse.ApplicationError
+                  }
+                  val errorMessage: String = Try(error.parseJson.convertTo[ErrorResponse])
+                    .map { e =>
+                      def logMsg = s"trigger-fired action '${rule.action}' failed to invoke with ${e.error}, ${e.code}"
```

Comment:
I'd prefer not to expose the `tid` here since that's an artifact of the current implementation and will go away. If there is a code, the error response should be sufficiently informative.

Reply:
Note this is a refactor. This is copied from the current code.

Reply:
You are right - this is just for the log message not the response logged in the activation. Sorry about the confusion.